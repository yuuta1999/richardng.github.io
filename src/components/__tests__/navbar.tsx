import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import Navbar from '../navbar'

afterEach(cleanup)

it('Navbar works with no transparency', () => {
  const { container } = render(<Navbar transparent={false}/>)

  expect(container.firstChild).toMatchSnapshot()
})


it('Navbar works with transparency', () => {
  const { container } = render(<Navbar transparent={true}/>)

  expect(container.firstChild).toMatchSnapshot()
})
