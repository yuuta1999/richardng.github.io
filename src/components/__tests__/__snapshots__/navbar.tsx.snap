// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar works with no transparency 1`] = `
.c0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.5rem;
  padding-right: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  background: var(--Theme-Navbar--Background);
  color: var(--Theme-Body--Text);
  border-bottom: 1px solid var(--Theme-Navbar--Border__onScroll);
  box-shadow: 1px 2px 18px var(--Theme-Navbar--BoxShadow__onScroll);
  -webkit-transition: padding 0.4s linear;
  transition: padding 0.4s linear;
}

.c0 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-flow: row nowrap;
    -ms-flex-flow: row nowrap;
    flex-flow: row nowrap;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0 .sc-ifAKCX {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .c1 {
    -webkit-flex-flow: nowrap;
    -ms-flex-flow: nowrap;
    flex-flow: nowrap;
  }

  .c0 .sc-bwzfXH {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .c0 .sc-bxivhb {
    display: none;
  }
}

<nav
  class="c0"
/>
`;

exports[`Navbar works with transparency 1`] = `
.c0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 1.5rem;
  padding-right: 0.5rem;
  padding-bottom: 1.5rem;
  padding-left: 0.5rem;
  background: transparent;
  color: var(--Theme-Body--Text);
  -webkit-transition: padding 0.4s linear;
  transition: padding 0.4s linear;
}

.c0 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-flow: row nowrap;
    -ms-flex-flow: row nowrap;
    flex-flow: row nowrap;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0 .sc-ifAKCX {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .c1 {
    -webkit-flex-flow: nowrap;
    -ms-flex-flow: nowrap;
    flex-flow: nowrap;
  }

  .c0 .sc-bwzfXH {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .c0 .sc-bxivhb {
    display: none;
  }
}

<nav
  class="c0"
/>
`;
